### DISCUSSION TITLE
How to run test?

### DISCUSSION THREAD

**Discussion by willhoh** [bookwyrm-social/bookwyrm#1940] [Q&A] [answered]
Hi, I have now tried several times to run pytest in development. But it doesn't do what I expect, even if I include an error on purpose. As I understood it, you have to type `./bw-dev pytest`. But I only get 
```
+ case "$CMD" in
+ execweb pytest --no-cov-on-fail
+ docker-compose exec web pytest --no-cov-on-fail
```
If I change pytest in bw-dev from `execweb` to `runweb` it works and I get the test I expect. Now I don't know if I'm doing something wrong or if there is a bug, hence the discussion here.

**Comment by mouse-reeve** [comment, marked as answer]
Oh sorry about that - I was using execweb for complicated reasons relating to external dependencies, but it only works in the web container is already running. You were right to switch from execweb to runweb, and I'll update the bw-dev command
